resolving/tarball-resolver/src/index.ts

```diff
@@ -1,17 +1,23 @@
-import { type PkgResolutionId, type ResolveResult } from '@pnpm/resolver-base'
+import { type PkgResolutionId, type ResolveResult, type TarballResolution } from '@pnpm/resolver-base'
 import { type FetchFromRegistry } from '@pnpm/fetching-types'
 
+export interface UrlResolveResult extends ResolveResult {
+  normalizedBareSpecifier: string
+  resolution: TarballResolution
+  resolvedVia: 'url'
+}
```

Comment:
I don't know whether this should be named `UrlResolveResult` or `TarballResolveResult`. The function name suggests "tarball" but the field `resolvedVia` suggests "url".

Reply:
TarballResolveResult sounds good to me

Reply:
https://github.com/pnpm/pnpm/pull/9544/commits/841c2fae983c43a97e6082dda3ce24577cbcb1a7